### ISSUE TITLE
Android action handlers in JS not triggered

### ISSUE THREAD

**Issue by sanmed** [phonegap/phonegap-plugin-push#590]
I am using 1.6 and have set the foreground to false in the payload. When the app is terminated and a push notification comes through, clicking on the buttons doesn't trigger the JS handlers. Here's what I see in the logcat log

02-08 13:17:49.117  5091  5091 D PushPlugin_BackgroundActionButtonHandler: BackgroundActionButtonHandler = Bundle[mParcelledData.dataSize=912]
02-08 13:17:49.117  5091  5091 D PushPlugin_BackgroundActionButtonHandler: not id = 0
02-08 13:17:49.119  5091  5091 V PushPlugin: sendExtras: caching extras to send at a later time.
02-08 13:17:49.124   204   809 D audio_hw_primary: select_devices: out_snd_device(2: speaker) in_snd_device(0: none)

I do log the entry point for the triggers in JS and don't see them in the logcat logs.

Is foreground set to false expected to work in the coldstart/in memory scenarios?

**Comment by macdonst** [comment]
@sanmed Yeah, if you kill your app the callback won't be invoked. Related to #158

**Comment by sanmed** [comment]
For what use cases does the foreground of false work? Does it work when the app is in a suspended state?

Any idea when the issue maybe resolved (for firing callbacks when the app is in a terminated state)

**Comment by macdonst** [comment]
@sanmed when the app is in the background and you set foreground to false on your action button the callback does get called. At least I see my callback get invoked when I attach a debugger to the app. 

As for how to fire a callback when the app is terminated, no set timetable. Ideally you should not terminate the app but I understand that users do kill apps. So pull requests are welcome :smile:

**Comment by sanmed** [comment]
Thanks Simon. Do you have any updates on this? We'd really like the interactive push notifications to work when the app is in the terminated state :) Any help on this is appreciated. Also, can you provide your contact info as I'd like to run by you a few questions I have? Thanks

**Comment by macdonst** [comment]
@sanmed nope, no updates on this. I haven't had time to work on that particular issue. Not sure if there is much we can do if somone force closes your app.

Re: contact info. I'm @macdonst on twitter, github and the Cordova slack channel so start there.

**Comment by ejazdragonflycreation** [comment]
passing data with action button, because when app is closed and notification is going to come and when we click on action button the callback is call but how to catch data with this click
